pkg/apis/kops/v1alpha2/conversion.go

```diff
@@ -419,7 +419,27 @@ func Convert_kops_ClusterSpec_To_v1alpha2_ClusterSpec(in *kops.ClusterSpec, out
 	if out.API.IsEmpty() {
 		out.LegacyAPI = nil
 	}
-	out.LegacyCloudProvider = string(in.GetCloudProvider())
+	if in.CloudProvider.AWS != nil {
```

Comment:
We don't support experimental cloud providers here (we don't have the cluster, and thus don't have the labels), but I think that's OK.